homeassistant/components/nextbus/coordinator.py

```diff
@@ -48,27 +48,63 @@ def has_routes(self) -> bool:
         """Check if this coordinator is tracking any routes."""
         return len(self._route_stops) > 0
 
+    async def async_shutdown(self) -> None:
+        """If there are no more routes, cancel any scheduled call, and ignore new runs."""
+        if self.has_routes():
+            return
+
+        await super().async_shutdown()
+
     async def _async_update_data(self) -> dict[str, Any]:
         """Fetch data from NextBus."""
 
-        _route_stops = set(self._route_stops)
-        self.logger.debug("Updating data from API. Routes: %s", str(_route_stops))
+        _stops_to_route_stops: dict[str, set[RouteStop]] = {}
+        for route_stop in self._route_stops:
+            _stops_to_route_stops.setdefault(route_stop.stop_id, set()).add(route_stop)
```

Comment:
For solutions I only ever need to create once, I tend to use `setdefault` rather than `defaultdict`, mostly just because I find it simpler. Since I’m not defaulting anywhere else, I’d prefer to keep it this way. 

Do you have strong feelings about changing it?

Reply:
My main reason would be to remove the `.setdefault` and just directly do `.add` which looks cleaner imo, but if you're not up for it I am not going to block it